server/apiRoutes/post.js

```diff
@@ -0,0 +1,80 @@
+const router = require('express').Router();
+// const { client, cacheDuration } = require('../redis');
+const { Post } = require('../db/index');
+
+// create an event
+router.put('/createPost', async (req, res, next) => {
+    try {
+        res.json(await Post.create({
+            ...req.body,
+            userId: req.user.id,
+            groupId: req.params.groupId,
+        }));
+    } catch (err) {
+        next(err);
+    }
+});
+
+// grab a single post i.e. when someone clicks on a post link to read comments
+router.get('/:postId', async (req, res, next) => {
+    try {
+        res.json(await Post.findOne({
+            where: {
+                id: req.params.postId
+            }
+        }));
+    } catch (err) {
+        next(err)
+    }
+})
+
+//get all posts associated with a user
+router.get('/userPosts', async (req, res, next) => {
```

Comment:
There should now be some isLoggedIn middleware in /utils/backend on the dev branch. please use that here.